### PR TITLE
qli chapter05

### PR DESCRIPTION
**Pull request by qianliok**

### CODE REVIEW

File: chapter05/qli/ex1.pl [hunk 1]
```diff
@@ -0,0 +1,7 @@
+#!/usr/bin/perl
+use warnings;
+use strict;
+
+print reverse <>;
```
Comment (codavex) [inline chapter05/qli/ex1.pl:5]: One line :+1: :smile:

File: chapter05/qli/ex2.pl [hunk 2]
```diff
@@ -0,0 +1,13 @@
+#!/usr/bin/perl
+use warnings;
+use strict;
+
+
+chomp( my @inputs = <STDIN> );
+
+print "1234567890"x5, "\n" ;
```
Comment (codavex) [inline chapter05/qli/ex2.pl:8]: I think this is nicer than the way I printed the ruler. :+1:

File: chapter05/qli/ex3.pl [hunk 3]
```diff
@@ -0,0 +1,19 @@
+#!/usr/bin/perl
+use warnings;
+use strict;
+use Scalar::Util qw ( looks_like_number );
+
+my $width;
+while ( !defined $width || !looks_like_number($width) ){
```
Comment (codavex) [inline chapter05/qli/ex3.pl:7]: Good validation

File: chapter05/qli/barney [hunk 4]
```diff
@@ -0,0 +1,3 @@
+hiya
+hello
+aloha
```
Comment (codavex) [inline chapter05/qli/barney:3]: Hola! :wave:

File: chapter05/qli/ex2.pl (hunk 2 above)
Comment (alexdeas) [inline chapter05/qli/ex2.pl:8]: Nope, @codavex used a much cooler method.

Comment (qianliok) [inline chapter05/qli/ex2.pl:8]: actually I was thinking to do something like  `((1..9)."0")x5`, but I couldn't figure out how to write it properly, so end up typing every number from 1 to 9. :smile:

Comment (pratyp) [inline chapter05/qli/ex2.pl:8]: Took me a while too but figured we could just do <code>((1..9),0)</code> :sweat_smile: 
Tried with >20 character string and see what it does?

File: chapter05/qli/ex3.pl (hunk 3 above)
Comment (pratyp) [inline chapter05/qli/ex3.pl:7]: :+1:

File: chapter05/qli/ex2.pl (hunk 2 above)
Comment (qianliok) [inline chapter05/qli/ex2.pl:8]: just found that pxd did this properly, e.g. `((1..9,0) x 5)`

Comment (codavex) [inline chapter05/qli/ex2.pl:8]: Dunno @alexdeas , we could argue over 'nice' vs 'cool' (:smile: vs :sunglasses: ?)

String repetition vs sequence operator, a loop and modulus maths..seems simpler this way?

Comment (qianliok) [inline chapter05/qli/ex2.pl:8]: Thanks very much @dpratyusha . Somehow Github doesn't auto refresh with the new comments. I found it in your code as well. Nice one! :+1:

**Pull request merged**